tests/test_multipart_operations.py

```diff
@@ -4,92 +4,282 @@
+        log.info(mp_response)
+        log.info("Trying to download multipart object and validating it")
+        c_scope_s3client.download_bucket_contents(
+            resp["bucket_name"], resp["results_dir"]
+        )
+        assert check_data_integrity(resp["origin_dir"], resp["results_dir"])
+        log.info("Both uploaded and downloaded data are identical")
+
+    def test_s3_multipart_object_list(self, c_scope_s3client, tmp_directories_factory):
+        """
+
+        Test multipart object list operations using BOTO s3:
+        1. Write multipart objects to the bucket
+        2. List objects from the bucket and verify it
+
+        """
+        log.info("Uploading multipart object")
+        resp = self.upload_multipart_object(c_scope_s3client, tmp_directories_factory)
+        obj_name = resp["object_names"][0]
+        log.info("Trying to complete multipart operation for the object")
+        mp_response = c_scope_s3client.complete_multipart_object_upload(
+            resp["bucket_name"],
+            obj_name,
+            resp[f"{obj_name}" + "_upload_id"],
+            resp["all_part_info"],
+        )
+        log.info(mp_response)
+        log.info("Multipart operation is completed")
+        log.info(f"Listing objects present in {resp['bucket_name']}")
+        listed_objs = c_scope_s3client.list_objects(resp["bucket_name"])
+        log.info(listed_objs)
+        assert set(resp["object_names"]).issubset(
+            set(listed_objs)
+        ), "All uploaded objects are not present in bucket"
+        log.info("Uploaded objects are present in bucket")
+
+    def test_s3_multipart_list_parts(self, c_scope_s3client, tmp_directories_factory):
+        """
+
+        Test multipart object list operations using BOTO s3:
+        1. Write multipart objects to the bucket
+        2. List parts objects from the bucket
 
-    # 5. Download the objects from the bucket and verify data integrity
-    s3_client.download_bucket_contents(bucket_name, results_dir)
-    assert check_data_integrity(origin_dir, results_dir)
+        """
+        log.info("Uploading multipart object")
+        resp = self.upload_multipart_object(c_scope_s3client, tmp_directories_factory)
+        obj_name = resp["object_names"][0]
+        log.info(f"Listing incomplete multipart uploads for the object {obj_name}")
+        part_resp = c_scope_s3client.list_uploaded_parts(
+            resp["bucket_name"], obj_name, resp[f"{obj_name}" + "_upload_id"]
+        )
+        assert (
+            len(part_resp["Parts"]) != 0
+        ), f"Failed to list parts of {resp['object_names'][0]} object"
+        log.info(part_resp)
+        log.info(f"Listing incomplete parts for {obj_name} completed successfully")
+        log.info("Trying to complete multipart operation for the object")
+        mp_response = c_scope_s3client.complete_multipart_object_upload(
+            resp["bucket_name"],
+            obj_name,
+            resp[f"{obj_name}" + "_upload_id"],
```

Comment:
f"{obj_name}" is equivalent to obj_name - see the other comment for suggestoins

Reply:
Fixed